hybridse/src/udf/default_defs/array_def.cc

```diff
@@ -0,0 +1,116 @@
+
+namespace hybridse {
+namespace udf {
+
+// =========================================================== //
+//      External Array Function Defines
+// =========================================================== //
+
+template <typename T>
+struct ArrayContains {
+    // udf registry types
+    using Args = std::tuple<ArrayRef<T>, Nullable<T>>;
+
+    using ParamType = typename DataTypeTrait<T>::CCallArgType;
+
+    // type binding, udf registry type -> function param type
+    // - bool/intxx/float/double -> bool/intxx/float/double
+    // - Timestamp/Date/StringRef -> Timestamp*/Date*/StringRef*
+    bool operator()(ArrayRef<T>* arr, ParamType v, bool is_null) {
+        // NOTE: array_contains([null], null) returns null
+        // this might not expected
+        for (uint64_t i = 0; i < arr->size; ++i) {
+            if constexpr (std::is_pointer_v<ParamType>) {
+                // null or same value returns true
+                if ((is_null && arr->nullables[i]) || (!arr->nullables[i] && *arr->raw[i] == *v)) {
+                    return true;
+                }
+            } else {
+                if ((is_null && arr->nullables[i]) || (!arr->nullables[i] && arr->raw[i] == v)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+};
+
+void SplitString(StringRef* str, StringRef* delimeter, ArrayRef<StringRef>* array) {
+    std::vector<absl::string_view> stats = absl::StrSplit(absl::string_view(str->data_, str->size_),
+                                                          absl::string_view(delimeter->data_, delimeter->size_));
+
+    array->size = stats.size();
+    if (!stats.empty()) {
+        v1::AllocManagedArray(array, stats.size());
+
+        for (size_t i = 0; i < stats.size(); ++i) {
+            auto s = stats[i];
+            if (s.empty()) {
+                array->raw[i]->size_ = 0;
+                array->nullables[i] = true;
+            } else {
+                char* buf = v1::AllocManagedStringBuf(s.size());
+                memcpy(buf, s.data(), s.size());
+                array->raw[i]->data_ = buf;
+                array->raw[i]->size_ = s.size();
+                array->nullables[i] = false;
+            }
+        }
+    }
+}
+
+// =========================================================== //
+//      UDF Register Entry
+// =========================================================== //
+void DefaultUdfLibrary::InitArrayUdfs() {
+    RegisterExternalTemplate<ArrayContains>("array_contains")
+        .args_in<bool, int16_t, int32_t, int64_t, float, double, Timestamp, Date, StringRef>()
+        .doc(R"(
+             @brief array_contains(array, value) - Returns true if the array contains the value.
+
+             Example:
+
+             @code{.sql}
+                 select array_contains([2,2], 2) as c0;
+                 -- output true
+             @endcode
+
+             @since 0.7.0
+             )");
+
+    RegisterExternal("split_array")
```

Comment:
it has the same functionality as `split`?

can I do `select split_array("2,1", ",")`?

Reply:
No, array type is not supported in storage. 

The two has same purpose, leaves later to handle the difference between array and list.